src/process/receipts.rs

```diff
@@ -0,0 +1,297 @@
+use std::collections::HashMap;
+use std::convert::TryFrom;
+
+use diesel::r2d2::{ConnectionManager, Pool};
+use diesel::{ExpressionMethods, JoinOnDsl, PgConnection, QueryDsl};
+use futures::join;
+use num_traits::cast::FromPrimitive;
+use tokio_diesel::AsyncRunQueryDsl;
+use tracing::{error, warn};
+
+use crate::models;
+use crate::schema;
+
+/// Saves receipts to database
+pub(crate) async fn process_receipts(
+    pool: &Pool<ConnectionManager<PgConnection>>,
+    receipts: Vec<&near_indexer::near_primitives::views::ReceiptView>,
+    block_height: u64,
+) {
+    let mut skipping_receipt_ids: Vec<near_indexer::near_primitives::hash::CryptoHash> = vec![];
+
+    let tx_hashes_for_receipts_via_outcomes: Result<
+        Vec<(String, String)>,
+        tokio_diesel::AsyncError,
+    > = schema::execution_outcome_receipts::table
+        .inner_join(
+            schema::receipts::table.on(
+                schema::execution_outcome_receipts::dsl::execution_outcome_receipt_id
+                    .eq(schema::receipts::dsl::receipt_id),
+            ),
+        )
+        .filter(
+            schema::execution_outcome_receipts::dsl::receipt_id.eq_any(
+                receipts
+                    .iter()
+                    .map(|r| r.receipt_id.to_string())
+                    .collect::<Vec<_>>(),
+            ),
+        )
+        .select((
+            schema::execution_outcome_receipts::dsl::receipt_id,
+            schema::receipts::dsl::transaction_hash,
+        ))
+        .load_async(&pool)
+        .await;
+
+    let tx_hashes_for_receipt_via_transactions: Result<
+        Vec<(String, String)>,
+        tokio_diesel::AsyncError,
+    > = schema::transactions::table
+        .filter(
+            schema::transactions::dsl::receipt_id.eq_any(
+                receipts
+                    .iter()
+                    .map(|r| r.receipt_id.to_string())
+                    .collect::<Vec<_>>(),
+            ),
+        )
+        .select((
+            schema::transactions::dsl::receipt_id,
+            schema::transactions::dsl::transaction_hash,
+        ))
+        .load_async(&pool)
+        .await;
+
+    let mut tx_hashes_for_receipts: HashMap<String, String> = HashMap::new();
+
+    if let Ok(result) = tx_hashes_for_receipts_via_outcomes {
```

Comment:
We should retry fetching instead of silently ignoring anything we have in the DB.

Reply:
But we can stuck in retries if there is nothing in DB and will never be there.